auction-server/src/opportunity/entities/opportunity_svm.rs

```diff
@@ -292,4 +295,28 @@ impl OpportunitySvm {
             OpportunitySvmProgram::Limo(_) => vec![],
         }
     }
+
+    pub fn get_permission_key(router: Pubkey, permission_account: Pubkey) -> PermissionKey {
+        PermissionKey::from([router.to_bytes(), permission_account.to_bytes()].concat())
+    }
+}
+
+impl From<OpportunitySvm> for api::OpportunityDelete {
+    fn from(val: OpportunitySvm) -> Self {
+        api::OpportunityDelete::Svm(api::OpportunityDeleteSvm::V1(api::OpportunityDeleteV1Svm {
+            chain_id:           val.chain_id.clone(),
+            permission_account: val.permission_account,
+            router:             val.router,
+            program:            val.program.into(),
+        }))
+    }
+}
+
+impl From<OpportunitySvmProgram> for api::ProgramSvm {
```

Comment:
Same here:
These conversions are lossy, so I prefer a specific method over using the From trait.
https://doc.rust-lang.org/std/convert/trait.From.html#when-to-implement-from

Reply:
I think this one makes more sense, but I removed the one for `OpportunityDelete`